examples/map/map_segment.py

```diff
@@ -0,0 +1,88 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demostration creating a segment of a particular map from
+transformed coordinates.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import NorthOffsetFrame
+from sunpy.data.sample import AIA_171_IMAGE
+
+###############################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+###############################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+###############################################################################
+# Let's then segment the data based on coordinates and create a boolean mask
+# where True indicates the segmented mask.
+segment = np.logical_and(all_hgs.lon < 10 * u.deg, all_hgs.lon > 0 * u.deg)
+
+###############################################################################
+# Let's plot the segment separately, we create a new map with the segment as the
+# mask.
+new_frame_map = sunpy.map.Map(
+    smap.data,
+    smap.meta,
+    mask=np.logical_not(segment))
+fig = plt.figure()
+ax = plt.subplot(projection=smap)
+new_frame_map.plot()
+plt.show()
+
+###############################################################################
+# We can perform various mathematical operations on the extracted segment such
+# as averaging the pixel values or finding the sum of the segment.
+new_frame_mask = np.ma.array(smap.data, mask=np.logical_not(segment))
```

Comment:
```suggestion
new_frame_mask = np.ma.array(smap.data, mask=segment)
```

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.ma.mean.html
As per numpy's masked arrays, the values marked `False` are considered right? So here we want only the segment and we'd have to consider everything else as True hence the `np.logical_not`.
Correct me if I'm wrong, I'm a bit confused here.